src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -68,10 +70,19 @@ default URLDownload getUrlDownload(String query) throws MalformedURLException, F
         }
 
         try (InputStream stream = getUrlDownload(query).asInputStream()) {
-            List<BibEntry> fetchedEntries = getParser().parseEntries(stream);
+            List<BibEntry> fetchedEntries = new ArrayList<>();
 
-            // Post-cleanup
-            fetchedEntries.forEach(this::doPostCleanup);
+            // check if there is anything to read since mEDRA '404 not found' returns nothing
+            PushbackInputStream pushbackInputStream = new PushbackInputStream(stream);
```

Comment:
You should wrap the stream in a Try With Resources Statement to make sure that the stream gets closed automatically on errors. And also please rename "b" to a more meaningful variable, e.g. readByte 
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
This is still not adressed. Needs to be as soon as our git issues are resolved ^^